src/main/java/mate/academy/spring/controller/UserController.java

```diff
@@ -0,0 +1,31 @@
+package mate.academy.spring.controller;
+
+import java.util.NoSuchElementException;
+import mate.academy.spring.mapper.DtoResponseMapper;
+import mate.academy.spring.model.User;
+import mate.academy.spring.model.dto.response.UserResponseDto;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+    private final UserService userService;
+    private final DtoResponseMapper<UserResponseDto, User> userDtoResponseMapper;
+
+    public UserController(UserService userService,
+                          DtoResponseMapper<UserResponseDto, User> userDtoResponseMapper) {
+        this.userService = userService;
+        this.userDtoResponseMapper = userDtoResponseMapper;
+    }
+
+    @GetMapping("/by-email")
+    public UserResponseDto findByEmail(@RequestParam String email) {
+        User user = userService.findByEmail(email)
+                .orElseThrow(NoSuchElementException::new);
```

Comment:
It's a good practice to include a meaningful error message when throwing an exception. For example: `() -> new EntityNotFoundException("User not found with email: " + email)`.